src/main/java/org/aerogear/kryptowire/KryptowireServiceImpl.java

```diff
@@ -19,15 +21,24 @@
     private static final Logger logger = Logger.getLogger(KryptowireServiceImpl.class.getName());
 
     private String apiKey;
-
     private String apiEndpoint;
+    private HttpClient httpclient;
 
-    public KryptowireServiceImpl(String apiEndpoint, String apiKey) {
-        if (this.apiEndpoint != null && !this.apiEndpoint.isEmpty() && !this.apiEndpoint.endsWith("/")) {
-            this.apiEndpoint += "/";
-        }
-        this.apiEndpoint = apiEndpoint;
+    public KryptowireServiceImpl(String apiKey, String apiEndpoint, HttpClient httpclient) {
         this.apiKey = apiKey;
+        this.apiEndpoint = apiEndpoint;
+        this.httpclient = httpclient;
+    }
+
+    public KryptowireServiceImpl(String apiEndpointParam, String apiKeyParam) {
+        this.httpclient = HttpClients.createDefault();
+        this.apiEndpoint = apiEndpointParam;
+        this.apiKey = apiKeyParam;
+        if (this.apiEndpoint != null) {
+            if (!this.apiEndpoint.isEmpty() && !this.apiEndpoint.endsWith("/")) {
+                this.apiEndpoint += "/";
+            }
+        }
     }
```

Comment:
Would you be better to call the other constructor from this one and add the logic to check the endpoint there, currently it's only being checked in this one. i.e. 

```
    public KryptowireServiceImpl(String apiEndpoint, String apiKey, HttpClient httpclient) {
        this.apiKey = apiKey;
        this.apiEndpoint = apiEndpoint;
        this.httpclient = httpclient;
        if (this.apiEndpoint != null) {
            if (!this.apiEndpoint.isEmpty() && !this.apiEndpoint.endsWith("/")) {
                this.apiEndpoint += "/";
            }
        }
    }

    public KryptowireServiceImpl(String apiEndpoint, String apiKey) {
        this(apiEndpoint, apiKey, HttpClients.createDefault());
    }
```

Reply:
I added a prepare method to handle this code on both constructors

Reply:
I see that, personally i think overloading the constructor and limiting the duplication is preferable that another method tat will only ever be called from the constructor.